src/Microsoft.OData.ModelBuilder/Vocabulary/IRecord.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.OData.ModelBuilder
+{
+	/// <summary>
+	/// Interface for clr types that can be converted into <see cref="EdmRecordExpression" />
+	/// </summary>
+	public interface IRecord
```

Comment:
should it be a Generic interface to accept the "CLR" type to convert, that "CLR" type should derive from a base type?

Reply:
Could you please clarify?

- Do we convert this into an abstract base class?
  Thought about this but it would just have been practically an interface ie an abstract class with just abstract methods...

- Do have a generic class of T for which we have implementations for the ClrTypeRecords ToEdmExpression()?
  I'd thought of and implemented something like this but it came out a bit complex for the task we wanted...

- You mean something else?